### ISSUE TITLE
[Cloud Code] Support role-based access control in lambda / handler

### ISSUE THREAD

**Issue by akiroz** [SkygearIO/skygear-SDK-JS#198]
Right now lambda / handler access can only be controlled using `authRequired` and `userRequired`. It would be nice to be able to control access by user role just like records.

**Comment by cheungpat** [comment]
I just want to add that this feature requires support from skygear-server.

**Comment by akiroz** [comment]
Couldn't we just query the `_user_role` table from the plugin?

**Comment by cheungpat** [comment]
> Couldn't we just query the _user_role table from the plugin?

Yes, but that would break the current separation between plugin and server. For example, if we implement it in the plugin side, the python cloud code would not support this feature.

This feature requires some design. Could you add this to SkygearIO/features where this feature will get more focus.
